frontend/src/App.tsx

```diff
@@ -1,19 +1,72 @@
-import './App.css';
-import Ready from './pages/ReadyPage/Ready';
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
 import CharacterSelection from './pages/CharacterSelectionPage/CharacterSelection';
+import Ready from './pages/ReadyPage/Ready';
 import StartPage from './pages/StartPage/StartPage';
 import SceneOnePage from './pages/SceneOnePage/SceneOnePage';
 import SceneTwoPage from './pages/SceneTwoPage/SceneTwoPage';
 import SceneThreePage from './pages/SceneThreePage/SceneThreePage';
-import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
-import SceneGoalPage from "./pages/SceneGoalPage/SceneGoalPage.tsx";
+import SceneGoalPage from "./pages/SceneGoalPage/SceneGoalPage";
+import { HeroDto, SaveGameDto } from "./types/types.ts";
 
 export default function App() {
+    const [saveGames, setSaveGames] = useState<SaveGameDto[]>([]);
+    const [heroes, setHeroes] = useState<HeroDto[]>([]);
+    const [error, setError] = useState<string | null>(null);
+    const [shouldUpdate, setShouldUpdate] = useState(false);
+
+    useEffect(() => {
+        const fetchDataAndUpdate = async () => {
+            try {
+                const [saveGamesResponse, heroesResponse] = await Promise.all([
+                    axios.get<SaveGameDto[]>('/api/savegames'),
+                    axios.get<HeroDto[]>('/api/heroes')
+                ]);
+
+                setSaveGames(saveGamesResponse.data);
+                setHeroes(heroesResponse.data);
+                setError(null);
+            } catch (error) {
+                const typedError = error as Error;
+                if (typedError.message) {
+                    setError(`Fehler beim Laden der Daten: ${typedError.message}`);
+                } else {
+                    setError('Fehler beim Laden der Daten');
+                }
+            }
+        };
+        fetchDataAndUpdate()
+            .catch((error) => {
+                console.error('Fehler beim Laden der Daten:', error);
+            });
```

Comment:
I think you don't need the `.catch` function here. Because you have already on the the function itself.

Reply:
you're right about the .catch in the useEffect. Since fetchDataAndUpdate already has its own try-catch block, any errors thrown will be caught there. Having an additional .catch on fetchDataAndUpdate() in useEffect is redundant. Thanks for pointing that out!